x/gov/types/v1/msgs.go

```diff
@@ -61,9 +61,9 @@ func (m MsgSubmitProposal) ValidateBasic() error {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, deposit.String())
 	}
 
-	// Check that either metadata or Msgs length is non nil.
-	if len(m.Messages) == 0 && len(m.Metadata) == 0 {
-		return sdkerrors.Wrap(types.ErrNoProposalMsgs, "either metadata or Msgs length must be non-nil")
+	// Check Msgs length is non nil.
```

Comment:
I see that this previously used the term "non nil" but I find that confusing b/c a non-nil `m.Messages` would return false here (i.e. if `len(m.Messages)=1).

```suggestion
	// Check if no Messages are proposed
```